api/tests/integration/infrastructure/repositories/campaign-repository_test.js

```diff
@@ -370,28 +370,56 @@ describe('Integration | Repository | Campaign', () => {
         });
       });
 
-      context('when some campaigns creatorId match the the given creatorId searched', () => {
+      context('when some campaigns creator firstName match the the given creatorName searched', () => {
 
         let filter;
 
         beforeEach(() => {
-          const creatorId1 = databaseBuilder.factory.buildUser({}).id;
-          const creatorId2 = databaseBuilder.factory.buildUser({}).id;
+          const creator1 = databaseBuilder.factory.buildUser({});
+          const creator2 = databaseBuilder.factory.buildUser({});
 
-          filter = { creatorId: creatorId1 };
+          filter = { creatorName: creator1.firstName };
 
           _.each([
-            { name: 'Maths L1', creatorId: creatorId1 },
-            { name: 'Maths L2', creatorId: creatorId2 },
-            { name: 'Chimie', creatorId: creatorId1 },
+            { name: 'Maths L1', creatorId: creator1.id },
+            { name: 'Maths L2', creatorId: creator2.id },
+            { name: 'Chimie', creatorId: creator1.id },
           ], (campaign) => {
             databaseBuilder.factory.buildCampaign({ ...campaign, organizationId });
           });
 
           return databaseBuilder.commit();
         });
 
-        it('should return the campaign with right creatorId', async () => {
+        it('should return the campaign with right creator firstName', async () => {
+          const { models: actualCampaignsWithReports } = await campaignRepository.findPaginatedFilteredByOrganizationIdWithCampaignReports({ organizationId, filter, page });
+
+          expect(_.map(actualCampaignsWithReports, 'name')).to.have.members(['Maths L1', 'Chimie']);
+        });
+      });
+
+      context('when some campaigns creator lastName match the the given creatorName searched', () => {
+
+        let filter;
+
+        beforeEach(() => {
+          const creator1 = databaseBuilder.factory.buildUser({});
+          const creator2 = databaseBuilder.factory.buildUser({});
+
+          filter = { creatorName: creator1.lastName };
+
+          _.each([
+            { name: 'Maths L1', creatorId: creator1.id },
+            { name: 'Maths L2', creatorId: creator2.id },
+            { name: 'Chimie', creatorId: creator1.id },
+          ], (campaign) => {
+            databaseBuilder.factory.buildCampaign({ ...campaign, organizationId });
+          });
+
+          return databaseBuilder.commit();
```

Comment:
Je suis toujours surpris de voir un return dans un BeforeEach (vu la capacité de JS à laisser des valeurs de retour être silencieusement ignorées)
Je propose de faire comme dans `api/tests/acceptance/application/organization-invitation-controller_test.js`
```suggestion
          await databaseBuilder.commit();
```
Peut être que ça causera des timeout, comme rencontré avec @chrc

Reply:
Le return est acceptable ici, mocha va attendre que la promesse soit résolue avant de commencer un test.